ops/testing.py

```diff
@@ -2586,6 +2604,82 @@ class NonAbsolutePathError(Exception):
     """
 
 
+@_copy_docstrings(model.Container)
+class _TestingContainerFilesystemFacade(Protocol):
+    """A Protocol class that represents a filesystem-related subset of operations of container."""
+
+    def pull(self, path: StrOrPath, *,
+             encoding: Optional[str] = 'utf-8') -> Union[BinaryIO, TextIO]:
+        ...
+
+    def push(self,
+             path: StrOrPath,
+             source: Union[bytes, str, BinaryIO, TextIO],
+             *,
+             encoding: str = 'utf-8',
+             make_dirs: bool = False,
+             permissions: Optional[int] = None,
+             user_id: Optional[int] = None,
+             user: Optional[str] = None,
+             group_id: Optional[int] = None,
+             group: Optional[str] = None):
+        ...
+
+    def list_files(self, path: StrOrPath, *, pattern: Optional[str] = None,
+                   itself: bool = False) -> List['FileInfo']:
+        ...
+
+    def push_path(self,
+                  source_path: Union[StrOrPath, Iterable[StrOrPath]],
+                  dest_dir: StrOrPath):
+        ...
+
+    def pull_path(self,
+                  source_path: Union[StrOrPath, Iterable[StrOrPath]],
+                  dest_dir: StrOrPath):
+        ...
+
+    def exists(self, path: str) -> bool:
+        ...
+
+    def isdir(self, path: str) -> bool:
+        ...
+
+    def make_dir(
+            self, path: str, *, make_parents: bool = False, permissions: Optional[int] = None,
+            user_id: Optional[int] = None, user: Optional[str] = None,
+            group_id: Optional[int] = None, group: Optional[str] = None):
+        ...
+
+    def remove_path(self, path: str, *, recursive: bool = False):
+        ...
+
+
+class _TestingContainerFilesystemProxy:
+    """A proxy providing unrestricted filesystem access for a specified container."""
+
+    def __init__(self, container: model.Container, harness: Harness[Any]):
+        self.__container = container
+        self.__harness = harness
+
+    def __getattr__(self, item: str):
+        if item not in _TestingContainerFilesystemFacade.__dict__ or item.startswith("_"):
+            raise AttributeError(f'object has no attribute {item!r}')
+        harness = self.__harness
+        container = self.__container
```

Comment:
Any reason to extract these to locals rather than just use eg `self._container` directly?

Reply:
No reason for that, I just updated the function to use `self._container` directly.